packages/core/src/com/communication.ts

```diff
@@ -685,8 +685,18 @@ export class Communication {
 
     private handleCallback(message: CallbackMessage): void {
         const rec = message.callbackId ? this.callbacks[message.callbackId] : null;
-        if (rec) {
-            message.error ? rec.reject(new Error(REMOTE_CALL_FAILED(message))) : rec.resolve(message.data);
+        if (rec && message.error) {
+            // If the error is not caught later, and logged to the console,
+            // it would be nice to indicate which environment the error came from.
+            // We shouldn't change the error message or the error name because
+            // those could be used by error-handling code. Fortunately, we can
+            // add metadata to the stack trace, and Chrome logs the entire stack
+            // property to the console. Unfortunately, Firefox doesn't.
+            const error = Object.assign(new Error(), message.error);
+            error.stack = REMOTE_CALL_FAILED(message.from, error.stack);
+            rec.reject(error);
```

Comment:
didn't we also discuss the error.type should be ``Remote${error.type}``?

Reply:
Yes. I've added that information to the stack trace instead, it's still visible in the console when you don't catch the error:

![image](https://user-images.githubusercontent.com/415252/106459628-d10d8f00-649a-11eb-90e6-d75e07ad05a5.png)

This allows us to always preserve the error name while still providing indication that this is a remote call.